test/action_controller/json_api/fields_test.rb

```diff
@@ -0,0 +1,57 @@
+require 'test_helper'
+
+module ActionController
+  module Serialization
+    class JsonApi
+      class FieldsTest < ActionController::TestCase
+        class FieldsTestController < ActionController::Base
+          class PostSerializer < ActiveModel::Serializer
+            type 'posts'
+            attributes :title, :body, :publish_at
+            belongs_to :author
+            has_many :comments
+          end
+
+          def setup_post
+            ActionController::Base.cache_store.clear
+            @author = Author.new(id: 1, first_name: 'Bob', last_name: 'Jones')
+            @comment1 = Comment.new(id: 7, body: 'cool', author: @author)
+            @comment2 = Comment.new(id: 12, body: 'awesome', author: @author)
+            @post = Post.new(id: 1337, title: 'Title 1', body: 'Body 1',
+                             author: @author, comments: [@comment1, @comment2],
+                             publish_at: '2020-03-16T03:55:25.291Z')
+            @comment1.post = @post
+            @comment2.post = @post
+          end
+
+          def render_fields_works_on_relationships
+            setup_post
+            render json: @post, serializer: PostSerializer, adapter: :json_api, fields: { posts: [:author] }
+          end
+        end
+
+        tests FieldsTestController
+
+        test 'fields works on relationships' do
```

Comment:
On a side note, I think we need some organization of the tests. 
maybe split them in to Unit, and Integration, and the further folders in Integration by topic of what's being tested.
Unit would follow the structure of `lib` - If this sounds like a good idea, lemme know -- this could cause conflicts with all the PRs, though. :-\

I just have a hard time finding things in `tests`